test/installation/assets/puppeteer/puppeteer-in-extension.js

```diff
@@ -0,0 +1,40 @@
+import puppeteer from 'puppeteer';
+import path from 'path';
+
+const pathToExtension = path.join(
+  process.cwd(),
+  'puppeteer-in-extension',
+  'out'
+);
+
+const browser = await puppeteer.launch({
+  headless: true,
+  args: [
+    `--disable-extensions-except=${pathToExtension}`,
+    `--load-extension=${pathToExtension}`,
+    '--silent-debugger-extension-api',
+  ],
+});
+
+try {
+  const port = process.argv[2];
+
+  const workerTarget = await browser.waitForTarget(
+    target =>
+      target.type() === 'service_worker' &&
+      target.url().endsWith('background.js')
+  );
+  const worker = await workerTarget.worker();
+
+  await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
Why do we need this?

Reply:
to wait for the service worker to be ready